prereise/gather/demanddata/bldg_electrification/floor_area_avg_slopes.py

```diff
@@ -0,0 +1,515 @@
+                    * puma_data_zone["frac_elec_sh_com_2019"]
+                )
+                * puma_data_zone["frac_in_zone"]
+            ).sum()
+            total_area_res = (
+                puma_data_zone["res_area_2019_m2"] * puma_data_zone["frac_in_zone"]
+            ).sum()
+            total_area_com = (
+                puma_data_zone["com_area_2019_m2"] * puma_data_zone["frac_in_zone"]
+            ).sum()
+            zone_floor_area = pd.concat(
+                [
+                    zone_floor_area,
+                    pd.DataFrame(
+                        {
+                            "total res": total_area_res,
+                            "total com": total_area_com,
+                            "heat": total_area_zone_heat_res + total_area_zone_heat_com,
+                            "% Heated": (
+                                (total_area_zone_heat_res + total_area_zone_heat_com)
+                                / (total_area_res + total_area_com)
+                            )
+                            * 100,
+                            "res heat": total_area_zone_heat_res,
+                            "com heat": total_area_zone_heat_com,
+                            "cool": total_area_zone_cool_res + total_area_zone_cool_com,
+                            "res cool": total_area_zone_cool_res,
+                            "com cool": total_area_zone_cool_com,
+                        },
+                        index=[zone],
+                    ),
+                ]
+            )
+    zone_floor_area.to_csv("./Profiles/result_stats/zone_floor_area_m2.csv")
+    return zone_floor_area
+
+
+def state_shp_overlay(state):
+    """Select load zones within a state
+
+    :param str state: abbrev. of state
+
+    :return: (*geopandas.GeoDataFrame*) state boundaries and load zones within it
+    """
+    state_shapefile = gpd.GeoDataFrame(
+        gpd.read_file(
+            os.path.join(
+                os.path.dirname(__file__), "shapefiles", "cb_2018_us_state_20m.shp"
+            )
+        )
+    )
+    state_shp = state_shapefile[state_shapefile["STUSPS"] == state]
+    zone_shp = gpd.GeoDataFrame(
+        gpd.read_file(
+            os.path.join(os.path.dirname(__file__), "shapefiles", "ba_area.shp")
+        )
+    ).to_crs("EPSG:4269")
+    zone_shp["area"] = zone_shp["geometry"].to_crs({"proj": "cea"}).area
+
+    zone_state = gpd.overlay(zone_shp, state_shp.to_crs("EPSG:4269"))
+    zone_state["area"] = zone_state["geometry"].to_crs({"proj": "cea"}).area
+    zone_state["area_frac"] = [
+        zone_state["area"][i]
+        / list(zone_shp[zone_shp["BA"] == zone_state["BA"][i]]["area"])[0]
+        for i in range(len(zone_state))
+    ]
+    zone_state.loc[zone_state["area_frac"] >= 0.99, "area_frac"] = 1
+    zone_state = zone_state.drop(zone_state[zone_state["area_frac"] <= 0.00001].index)
+
+    return zone_state
+
+
+def main_plots(iso_list):
```

Comment:
We are doing calculations and several plots all together in this main plot function. It would be great if calculations and plotting can be separated into different functions/modules and each function only does one thing, i.e. calculate/process one piece of data for a specific plot or plot one figure using the constructed data from other functions. Besides, it is always good to have keyword arguments for a plotting function so that the user could choose to save the figure or not. [Here](https://github.com/Breakthrough-Energy/PostREISE/tree/develop/postreise/plot) are some existing plotting functions in the codebase for references.